### PR TITLE
[BugFix] Turn off automatic partition creation when specifying partitions

### LINKED ISSUE #21222: [load auto create part] insert into temporary partition but create normal partition
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
CREATE TABLE `duplicate_table_with_null` ( `k1`  date, `k2`  datetime,`k3`  char(20), `k4`  varchar(20), `k5`  boolean, `k6`  tinyint, `k7`  smallint, `k8`  int, `k9`  bigint, `k10` largeint, `k11` float, `k12` double, `k13` decimal(27,9)) DUPLICATE KEY(`k1`, `k2`, `k3`, `k4`, `k5`) PARTITION BY date_trunc('day', k2) DISTRIBUTED BY HASH(`k1`, `k2`, `k3`) PROPERTIES (  "replication_num" = "3", "storage_format" = "v2");
insert into duplicate_table_with_null values('2020-01-01', '2020-01-01 12:30:30', 'asfgrgte', 'wergergqer', 0, 11, 111, 1111, 11111, 111111, 11.11, 111.111, 1111.1111),('2021-01-01', '2021-01-01 12:30:30', 'asfgrgtf', 'wergergqer', 1, 22, 222, 2222, 22222, 222222, 22.22, 222.222, 2222.2222),('2022-01-01', '2022-01-01 12:30:30', 'asfgrgtg', 'wergergqes', 0, 33, 333, 3333, 33333, 333333, 33.33, 333.333, 3333.3333),('2023-01-01', '2023-01-01 12:30:30', 'asfgrgth', 'wergergqet', 1, 44, 444, 4444, 44444, 444444, 44.44, 444.444, 4444.4444);

show partitions from duplicate_table_with_null order by PartitionName;
alter table duplicate_table_with_null add TEMPORARY partitions START ("2022-02-03") END ("2022-02-06") EVERY (INTERVAL 1 day);
INSERT INTO duplicate_table_with_null TEMPORARY PARTITION tp20220203 VALUES ('2022-02-03', '2022-02-05 12:32:12', 'asfgrgte', 'wergergqer', 0, 11, 111, 1111, 11111, 111111, 11.11, 111.111, 1111.1111);
show partitions from duplicate_table_with_null order by PartitionName;
```

### Expected behavior (Required)
The number of partitions of the table remains unchanged
### Real behavior (Required)
The number of partitions of the table changes

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
mysql> select current_version();
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-3.0-RELEASE 9cc2e97 |
+----------------------------+
1 row in set (0.01 sec)

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #21222

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java [hunk 1]
```diff
@@ -188,9 +188,15 @@ public ExecPlan plan(InsertStmt insertStmt, ConnectContext session) {
             if (insertStmt.getTargetTable() instanceof OlapTable) {
                 OlapTable olapTable = (OlapTable) insertStmt.getTargetTable();
 
-                dataSink = new OlapTableSink((OlapTable) insertStmt.getTargetTable(), olapTuple,
-                        insertStmt.getTargetPartitionIds(), canUsePipeline, olapTable.writeQuorum(),
-                        olapTable.enableReplicatedStorage(), nullExprInAutoIncrement);
+                boolean enableAutomaticPartition;
+                if (insertStmt.getTargetPartitionNames() != null) {
```
Comment (meegoo) [inline fe/fe-core/src/main/java/com/starrocks/sql/InsertPlanner.java:192]: suggest add comment

File: fe/fe-core/src/main/java/com/starrocks/sql/UpdatePlanner.java [hunk 2]
```diff
@@ -102,9 +102,11 @@ public ExecPlan plan(UpdateStmt updateStmt, ConnectContext session) {
                 for (Partition partition : table.getPartitions()) {
                     partitionIds.add(partition.getId());
                 }
+                OlapTable olapTable = (OlapTable) table;
                 DataSink dataSink =
-                        new OlapTableSink(((OlapTable) table), olapTuple, partitionIds, ((OlapTable) table).writeQuorum(),
-                                ((OlapTable) table).enableReplicatedStorage(), updateStmt.nullExprInAutoIncrement());
+                        new OlapTableSink(olapTable, olapTuple, partitionIds, olapTable.writeQuorum(),
```
Comment (meegoo) [inline fe/fe-core/src/main/java/com/starrocks/sql/UpdatePlanner.java:107]: broker load can also specify partition

File: fe/fe-core/src/main/java/com/starrocks/load/loadv2/LoadingTaskPlanner.java [hunk 3]
```diff
@@ -304,7 +305,8 @@ public void buildShufflePlan(TUniqueId loadId, List<List<TBrokerFileStatus>> fil
         // 4. Olap table sink
         List<Long> partitionIds = getAllPartitionIds();
         OlapTableSink olapTableSink = new OlapTableSink(table, tupleDesc, partitionIds, true,
-                table.writeQuorum(), table.enableReplicatedStorage(), checkNullExprInAutoIncrement());
+                table.writeQuorum(), table.enableReplicatedStorage(),
```
Comment (meegoo) [inline fe/fe-core/src/main/java/com/starrocks/load/loadv2/LoadingTaskPlanner.java:308]: broker load can also specify partition

File: fe/fe-core/src/main/java/com/starrocks/planner/StreamLoadPlanner.java [hunk 4]
```diff
@@ -175,7 +175,8 @@ public TExecPlanFragmentParams plan(TUniqueId loadId) throws UserException {
 
         List<Long> partitionIds = getAllPartitionIds();
         OlapTableSink olapTableSink = new OlapTableSink(destTable, tupleDesc, partitionIds, writeQuorum,
-                destTable.enableReplicatedStorage(), scanNode.nullExprInAutoIncrement());
+                destTable.enableReplicatedStorage(), scanNode.nullExprInAutoIncrement(),
+                destTable.supportedAutomaticPartition());
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/planner/StreamLoadPlanner.java:179]: stream load can specify partition

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**